backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTable.java

```diff
@@ -39,31 +41,60 @@ public interface CreateTableFunction {
 
     @FunctionalInterface
     public interface RequireRowLevelSecurity {
-        PostgresTable supportsRowLevelSecurity(boolean rowLevelSecurityEnabled);
+        FinalStage supportsRowLevelSecurity(boolean rowLevelSecurityEnabled);
 
-        default PostgresTable disableRowLevelSecurity() {
+        default FinalStage disableRowLevelSecurity() {
             return supportsRowLevelSecurity(false);
         }
 
-        default PostgresTable supportsRowLevelSecurity() {
+        default FinalStage supportsRowLevelSecurity() {
             return supportsRowLevelSecurity(true);
         }
     }
 
+    public static class FinalStage {
+        private final String tableName;
+        private final boolean supportsRowLevelSecurity;
+        private final Function<DSLContext, DDLQuery> createTableStepFunction;
+        private final ImmutableList.Builder<String> additionalConstraints;
```

Comment:
https://github.com/apache/james-project/pull/1849#discussion_r1424833536 If the PostgresIndex is acceptable, we don't need `additionalConstraints` here. 
If it is still necessary, We should rename it to -> `additionalAlterQueries`, it will be used in same cases that we want to use native sql

Reply:
Following the unique index adoption, I would split the `additionalAlterQueries` support into another PR. I think it still brings value as jOOQ DSL does not support all the cases.

Reply:
As long as we keep a fluent API for index creation I am fine

Reply:
> I would split the additionalAlterQueries support into another PR

https://github.com/apache/james-project/pull/1852